tests/binary-cache-build-remote.sh

```diff
@@ -0,0 +1,13 @@
+source common.sh
+
+clearStore
+clearCacheCache
+
+# Fails without remote builders
+(! nix-build --store "file://$cacheDir" dependencies.nix)
+
+# Succeeds with default store as build remote.
+nix-build --store "file://$cacheDir" --builders 'auto - - 1 1' -j0 dependencies.nix
```

Comment:
Maybe check here that the result actually exists in `$cacheDir`? E.g. `nix path-info --store "file://$cacheDir" $outPath` or something like that.